### PR TITLE
RudderStack IAM role for AWS-based destinations

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Added steps on creating a RS IAM role for role-based auth in all the AWS destinations supported by RudderStack.
> Added section on Workspace ID
> Added cross-references to the IAM doc in all the AWS destinations with the updated destination settings.

## Type of documentation

- [x] New documentation
- [x] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/destinations/aws-iam-role-for-rudderstack.mdx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+To set up a new RudderStack IAM role, follow these steps:
+
+1. Sign in to your AWS Management Console and open the <a href="https://console.aws.amazon.com/iam/">IAM console</a>.
+2. In the left navigation pane, click **Roles** followed by **Create role**.
+3. Under **Trusted entity type**, select **AWS account**, as shown:
+
+<img src="../assets/destinations/aws-role-1.png" alt="Setting up AWS IAM Role for RudderStack" />
+
+4. Select **Another AWS account** and under **Account ID**, enter `422074288268`, the account ID associated with RudderStack.
+5. Under **Options** check **Require external ID** and enter your <Link to="/resources/glossary/#workspace-token">workspace token</Link> as the **External ID**.
```
Comment (koladilip) [inline docs/destinations/aws-iam-role-for-rudderstack.mdx:31]: We are using WorkspaceId not the token

File: docs/destinations/aws-iam-role-for-rudderstack.mdx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+6. Optionally, you can choose to enable **Require MFA** based on your organization's security policies and requirements. Enabling this setting restricts the role only to the users who sign in with <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_mfa.html">multi-factor authentication (MFA)</a>.
```
Comment (koladilip) [inline docs/destinations/aws-iam-role-for-rudderstack.mdx:32]: This we can't support currently

File: docs/destinations/aws-iam-role-for-rudderstack.mdx [hunk 3]
```diff
@@ -0,0 +1,70 @@
+---
+title: "RudderStack IAM Role for AWS-based Destinations"
+description: >-
+  Step-by-step guide on creating a RudderStack IAM role for authenticating AWS-based destinations.
+---
+
+This guide walks you through the steps for creating a RudderStack IAM role. This role will be used for authenticating RudderStack in the following AWS-based destinations:
+
+- <Link to="/destinations/streaming-destinations/amazon-s3/">S3</Link>
```
Comment (koladilip) [inline docs/destinations/aws-iam-role-for-rudderstack.mdx:9]: S3 data lake too

File: docs/destinations/aws-iam-role-for-rudderstack.mdx (hunk 1 above)
Comment (ameypv-rudder) [reply docs/destinations/aws-iam-role-for-rudderstack.mdx:31]: yes Dilip - will make this change. Had added this before our discussion.

File: docs/destinations/aws-iam-role-for-rudderstack.mdx (hunk 2 above)
Comment (ameypv-rudder) [reply docs/destinations/aws-iam-role-for-rudderstack.mdx:32]: thanks for this note - i will remove this step.

Comment (koladilip) [reply docs/destinations/aws-iam-role-for-rudderstack.mdx]: image
We need to correct the image too

Comment (ameypv-rudder) [reply docs/destinations/aws-iam-role-for-rudderstack.mdx]: this is corrected.

File: docs/resources/glossary.mdx [hunk 4]
```diff
@@ -200,6 +200,12 @@ When sending your events to a data warehouse via RudderStack, you don't need to
 
 Refer to the <Link to="/destinations/warehouse-destinations/warehouse-schema/">Warehouse Schema</Link> guide for more details.
 
+## Workspace ID
+
+RudderStack uses the workspace ID for tracking the data internally.
+
+This ID is required when setting up an IAM role for RudderStack for authenticating all the AWS-based destinations. For more information, refer to the <Link to="/destinations/aws-iam-role-for-rudderstack/">Creating a AWS IAM Role for RudderStack</Link> guide.
```
Comment (ishachopra0401) [inline docs/resources/glossary.mdx:207]:
```suggestion
This ID is required when setting up an IAM role for RudderStack for authenticating all the AWS-based destinations. For more information, refer to the <Link to="/destinations/aws-iam-role-for-rudderstack/">Creating an AWS IAM Role for RudderStack</Link> guide.
```

**Review by ishachopra0401 — APPROVED** [review]

**Pull request merged**